### PR TITLE
User story 18

### PR DESCRIPTION
**Pull request by adumortier**
- added the model methods and the tests for the most and least popular `items`
- changed the items index view page to include these `items`

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -25,4 +25,12 @@ def no_orders?
     item_orders.empty?
   end
 
+  def self.most_popular_items
+    Item.joins(:item_orders).group("items.id").order("sum(item_orders.quantity) desc").limit(5).pluck(:name)
+  end
+
```
Comment (cgaddis36) [inline app/models/item.rb:31]: Active record logic looks sound

File: spec/features/items/index_stats_spec.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      address: '112 Elm St.',
+      city: 'Cambridge',
+      state: 'MA',
+      zip: '02139'
+    }
+
+    # order1 has items : item1, item3, item5, item7
+    # order2 has items : item2, item4, item6, item1
+    # total quantities ordered :
+    # item3 = 100
```
Comment (cgaddis36) [inline spec/features/items/index_stats_spec.rb:60]: Need to delete comments

File: spec/features/items/index_stats_spec.rb [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    item_order6 = ItemOrder.create!(item_order6_info)
+    
+    item_order7_info = {order_id: @order1.id, item_id: @item_7.id, price: @item_7.price, quantity: 54 }
+    item_order7 = ItemOrder.create!(item_order7_info)
+
+    item_order8_info = {order_id: @order2.id, item_id: @item_1.id, price: @item_1.price, quantity: 35 }
+    item_order8 = ItemOrder.create!(item_order8_info)
+
+    # most_popular = Item.joins(:item_orders).group("items.id").order("sum(item_orders.quantity) desc").limit(5).pluck(:name)
+    # least_popular = Item.joins(:item_orders).group("items.id").order("sum(item_orders.quantity)").limit(5).pluck(:name)
```
Comment (cgaddis36) [inline spec/features/items/index_stats_spec.rb:96]: More comments

**Review by cgaddis36 — APPROVED** [review]

File: app/models/item.rb [hunk 4]
```diff
@@ -25,4 +25,12 @@ def no_orders?
...
+  def self.least_popular_items
+    Item.joins(:item_orders).group("items.id").order("sum(item_orders.quantity)").limit(5).pluck(:name)
+  end
+
```
Comment (TravisBorgsmiller) [inline app/models/item.rb:35]: Good job joining with the item_orders table

**Review by TravisBorgsmiller — APPROVED** [review]

**Pull request merged**